Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Storage/Queries/Person.hs

```diff
@@ -86,30 +86,53 @@ findAllDriversWithInfoAndVehicle ::
   Maybe Text ->
   m [(Person, DriverInformation, Maybe Vehicle)]
 findAllDriversWithInfoAndVehicle merchantId limitVal offsetVal mbVerified mbEnabled mbBlocked mbSubscribed mbSearchPhoneDBHash mbVehicleNumberSearchString = do
+  drivers <- getDriversByAttributes merchantId mbSearchPhoneDBHash
+  driversInfo <- getDriversInfoByAttributes mbVerified mbEnabled mbBlocked mbSubscribed drivers
+  vehicles <- getVehiclesByNumber mbVehicleNumberSearchString driversInfo
+  let linkedDrivers = linkDriverWithVehicle drivers driversInfo vehicles
+  let result = case mbVehicleNumberSearchString of
+        Nothing -> linkedDrivers
+        Just _ -> filter (isJust . getVehicle) linkedDrivers
+  return $ Esq.limitOffset limitVal offsetVal result
+  where
+    getVehicle (_, _, vehicle) = vehicle
+
+linkDriverWithVehicle :: [Person] -> [DriverInformation] -> [Vehicle] -> [(Person, DriverInformation, Maybe Vehicle)]
+linkDriverWithVehicle drivers driversInfo vehicles = do
+  let personHM = buildPersonHashMap drivers
+      vehicleHM = buildVehicleHashMap vehicles
+   in mapMaybe (mapDriverWithVehicle personHM vehicleHM) driversInfo
+
+mapDriverWithVehicle :: HashMap.HashMap Text Person -> HashMap.HashMap Text Vehicle -> DriverInformation -> Maybe (Person, DriverInformation, Maybe Vehicle)
+mapDriverWithVehicle driverHM vehicleHM info = do
+  let driverId = info.driverId.getId
+  person <- HashMap.lookup driverId driverHM
+  let vehicle = HashMap.lookup driverId vehicleHM
+  Just (person, info, vehicle)
+
+getDriversByAttributes :: Transactionable m => Id Merchant -> Maybe DbHash -> m [Person]
+getDriversByAttributes merchantId mbSearchPhoneDBHash = do
```

Comment:
can we add `limit`, `offset` here ?

Reply:
No, if driver is searched by vehicle number, he could be missed if we add limit and offset here

Reply:
Yes, I also thought of doing that but we'll have a lot of checks in that case plus there will be concept of priority too. Let's say 
1. Only `vehicleNumber` is provided - need to make `vehicle` table as the initial set
2. Only `phoneNumber` is provided - `person` table as initial set
3. Both given -> Any order is fine
4. Both not given and `mbSubscribed` is given -> `driverInformation` as initial set

Similarly there can be other cases to check.
For now, my assumption was that this query is mostly used to search by `phoneNumber` and hence I didn't want to complicate or mess up the coding to optimize it for cases which doesn't occur much. This can be changed though, if we have high number of queries with `vehicleNumber`